include/verilated_timing.h

```diff
@@ -239,6 +239,91 @@ class VlTriggerScheduler final {
     }
 };
 
+//=============================================================================
+// VlDynamicTriggerScheduler is used for cases where triggers cannot be statically referenced and
+// evaluated. Coroutines that make use of this scheduler must adhere to a certain procedure:
+// __Vtrigger = 0;
+// <locals and inits required for trigger eval>
+// while (!__Vtrigger) {
+//     co_await __VdynSched.evaluation();
+//     <pre updates>;
+//     __Vtrigger = <trigger eval>;
+//     [optionally] co_await __VdynSched.postUpdate();
+//     <post updates>;
+// }
+// co_await __VdynSched.resumption();
+// The coroutines get resumed at trigger evaluation time, evaluate their local triggers, optionally
+// await the post update step, and if the trigger is set, await proper resumption in the 'act' eval
+// step.
+
+class VlDynamicTriggerScheduler final {
+    // TYPES
+    using VlCoroutineVec = std::vector<VlCoroutineHandle>;
+
+    // MEMBERS
+    VlCoroutineVec m_suspended;  // Suspended coroutines awaiting trigger evaluation
+    VlCoroutineVec
+        m_evaluated;  // Coroutines whose are currently being evaluated (only used in evaluate())
+    VlCoroutineVec m_triggered;  // Coroutines whose triggers were set, and are awaiting resumption
+    VlCoroutineVec m_post;  // Coroutines awaiting the post update step (only relevant for triggers
+                            // with destructive post updates, e.g. named events)
+
+    // METHODS
+    auto awaitable(VlCoroutineVec& queue, const char* filename, int linenum) {
+        struct Awaitable {
+            VlCoroutineVec& suspended;  // Coros waiting on trigger
+#ifdef VL_DEBUG
+            const char* filename;
+            int linenum;
+#endif
```

Comment:
Make a class with just this, and it's empty when no debug, so then all the other ifdefs can go away.

Reply:
https://github.com/verilator/verilator/pull/3652